common/src/main/java/io/netty/util/concurrent/GlobalEventExecutor.java

```diff
@@ -104,15 +104,11 @@ Runnable takeTask() {
     }
 
     private void fetchFromScheduledTaskQueue() {
-        if (hasScheduledTasks()) {
-            long nanoTime = AbstractScheduledEventExecutor.nanoTime();
-            for (;;) {
-                Runnable scheduledTask = pollScheduledTask(nanoTime);
-                if (scheduledTask == null) {
-                    break;
-                }
-                taskQueue.add(scheduledTask);
-            }
+        long nanoTime = AbstractScheduledEventExecutor.nanoTime();
+        Runnable scheduledTask = pollScheduledTask(nanoTime);
+        while (scheduledTask != null) {
+            taskQueue.add(scheduledTask);
```

Comment:
I was wondering that when I made the change. It looks to me, that the GlobalEventExecutor uses a LinkedBlockingQueue with the default capacity (max integer). So I assume there was not as much of a concern for a task being rejected by the queue as in the SingleThreadEventExecutor.